src/pages/ThorChainLP/queries/hooks/useSendThorTx.tsx

```diff
@@ -0,0 +1,458 @@
+import { ExternalLinkIcon } from '@chakra-ui/icons'
+import { Link, Text, useToast } from '@chakra-ui/react'
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import {
+  CONTRACT_INTERACTION,
+  type FeeDataEstimate,
+  FeeDataKey,
+} from '@shapeshiftoss/chain-adapters'
+import { SwapperName } from '@shapeshiftoss/swapper'
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { useQuery } from '@tanstack/react-query'
+import dayjs from 'dayjs'
+import { useCallback, useMemo, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { reactQueries } from 'react-queries'
+import { selectInboundAddressData } from 'react-queries/selectors'
+import { getAddress, zeroAddress } from 'viem'
+import type { SendInput } from 'components/Modals/Send/Form'
+import { estimateFees, handleSend } from 'components/Modals/Send/utils'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { getTxLink } from 'lib/getTxLink'
+import { fromBaseUnit, toBaseUnit } from 'lib/math'
+import { assertUnreachable, isToken } from 'lib/utils'
+import { assertGetThorchainChainAdapter } from 'lib/utils/cosmosSdk'
+import {
+  assertGetEvmChainAdapter,
+  buildAndBroadcast,
+  createBuildCustomTxInput,
+} from 'lib/utils/evm'
+import { THORCHAIN_POOL_MODULE_ADDRESS } from 'lib/utils/thorchain/constants'
+import { depositWithExpiry } from 'lib/utils/thorchain/routerCalldata'
+import { useGetEstimatedFeesQuery } from 'pages/Lending/hooks/useGetEstimatedFeesQuery'
+import { THORCHAIN_SAVERS_DUST_THRESHOLDS_CRYPTO_BASE_UNIT } from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+import {
+  selectAccountNumberByAccountId,
+  selectAssetById,
+  selectFeeAssetByChainId,
+  selectSelectedCurrency,
+} from 'state/slices/selectors'
+import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
+import { useAppSelector } from 'state/store'
+
+type ThorfiActionBase = 'ThorchainLp' | 'ThorchainLending' | 'ThorchainSavers'
+// Since there are different vernaculars (borrow/repay, deposit/withdraw) depending on the Thorfi action, we simplyfi it as enter/exit
+// Note, THORChain swaps *are* part of THORFi but there's no notion of enter/exit for them really
+type ThorfiAction = `${ThorfiActionBase}Enter` | `${ThorfiActionBase}Exit` | 'ThorchainSwap'
+
+type Props = {
+  amountCryptoBaseUnit: string | undefined
+  dustAmountCryptoBaseUnit: string | undefined
+  assetId: AssetId
+  // TODO(gomes): we should build this here too, but until this is properly implemented for all domains we're just passing it
```

Comment:
:+1:

Reply:
may still be useful to be an input to allow the component to more easily pass the necessary construction inputs or just the memo as returned from the quote endpoint, though I can see those memo construction helpers be unified as well where necessary

Reply:
798a19c152